Sofa/framework/Simulation/Core/src/sofa/simulation/MappingGraph.h

```diff
@@ -0,0 +1,85 @@
+/******************************************************************************
+*                 SOFA, Simulation Open-Framework Architecture                *
+*                    (c) 2006 INRIA, USTL, UJF, CNRS, MGH                     *
+*                                                                             *
+* This program is free software; you can redistribute it and/or modify it     *
+* under the terms of the GNU Lesser General Public License as published by    *
+* the Free Software Foundation; either version 2.1 of the License, or (at     *
+* your option) any later version.                                             *
+*                                                                             *
+* This program is distributed in the hope that it will be useful, but WITHOUT *
+* ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or       *
+* FITNESS FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License *
+* for more details.                                                           *
+*                                                                             *
+* You should have received a copy of the GNU Lesser General Public License    *
+* along with this program. If not, see <http://www.gnu.org/licenses/>.        *
+*******************************************************************************
+* Authors: The SOFA Team and external contributors (see Authors.txt)          *
+*                                                                             *
+* Contact information: contact@sofa-framework.org                             *
+******************************************************************************/
+#pragma once
+
+#include <sofa/simulation/config.h>
+#include <sofa/core/BaseMapping.h>
+#include <sofa/core/objectmodel/BaseContext.h>
+
+#include <queue>
+#include <unordered_set>
+
+namespace sofa::simulation
+{
+
+enum class SOFA_SIMULATION_CORE_API MappingGraphDirection : std::uint8_t
+{
+    TOP_DOWN,
```

Comment:
I'm not sure in a Graph there is the concept of TOP and DOWN which sound much like appropriate for a hierarchy. Forward and backward (which refer to the orientation of the arrows in a DAG sound more approriate)

Reply:
copy that 👍